### PR TITLE
V2.2

### LINKED ISSUE #16: custom fields for assignments does not work correctly
At line 378, the part of PowerShell code needs to be changed from: $_.customFields to $assignment.customFields. 
Please verify and test

### LINKED ISSUE #17: support base64 string for certificate
In the readme now states that the PowerShell needs to be edited whenever using cloud agent. 
The same logic can be used as in the part of the AccessToken. 
Use the variable $certificateBase64 to determine what you need to do. 
Possible solution could be: 
if (-not[string]::IsNullOrEmpty($certificateBase64)) {    
    $dataset = $datasetJson.content | ConvertFrom-Json
}
elseif (-not [string]::IsNullOrEmpty($certificatePathertificatePath)) {
    $datasetCorrected = [Text.Encoding]::UTF8.GetString([Text.Encoding]::GetEncoding(28591).GetBytes($datasetJson.content))
    $dataset = $datasetCorrected | ConvertFrom-Json
}
else {
    Throw "No certificate configured"
}

please test and verify

### PR DESCRIPTION
**Pull request by Gsichtman**
Apply feedback from mouki9.

- $assigment.customFields corrected
- mod mapping. Complex mapping on field to check if object exists
-  Fix diactric issue for cert.pfx

### CODE REVIEW

File: departments.ps1 [hunk 1]
```diff
@@ -244,7 +244,7 @@ Returns the raw JSON data containing all workers from ADP Workforce
 
             do {
                 $result = $null
-                $urlOffset = $Url + "?$" + "skip=$skip&$" + "top=$take"
+                $urlOffset = $Url + "?$" + "skip=$skip&$" + "top=$take&$" + "filter=MUT_INK_SOORT ne 31 and MUT_INK_SOORT ne 63 and MUT_INK_SOORT ne 62 and MUT_INK_SOORT ne 53 and /workers/workerDates/terminationDate gt $($filterDate)"
```
Comment (rhouthuijzen) [inline departments.ps1]: The variable $filterDate is not configured in the script. Is this something that should be configured in the configuration.json?

I have expirense with this source connector. Is $filterDate something that should be added to the person script?


Another question, is the line below something that applies to all customers or only for one?

"filter=MUT_INK_SOORT ne 31 and MUT_INK_SOORT ne 63 and MUT_INK_SOORT ne 62 and MUT_INK_SOORT ne 53"

File: mapping.json [hunk 2]
```diff
@@ -1,232 +1,203 @@
...
+            "value": "function getManagerExternalId() {\r\n    let managerExternalId = \"\";\r\n    let managerItemID = \"\";\r\n\r\n    if (typeof sourceContract.manager != null && sourceContract.manager != \"\" && sourceContract.manager != undefined && sourceContract.manager) {\r\n\r\n        if ((sourceContract.manager.itemID).indexOf(\"HOOFD\") !== -1) {\r\n            managerExternalId = sourceContract.manager.workerID.idValue;\r\n        } else if ( typeof sourceContract.reportsTo != null && sourceContract.reportsTo != \"\" && sourceContract.reportsTo != undefined) {\r\n            if ((sourceContract.reportsTo).length >= 1) {\r\n                for (i = 0; i < (sourceContract.reportsTo).length; i++) {\r\n                    managerItemID = sourceContract.reportsTo[i].itemID;\r\n                    if ((managerItemID).indexOf(\"HOOFD\") !== -1) {\r\n                        managerExternalId = sourceContract.reportsTo[i].workerID.idValue;\r\n                        break;\r\n                    }\r\n                }\r\n\r\n            }\r\n        }\r\n    }\r\n\r\n\r\n    return managerExternalId;\r\n}\r\ngetManagerExternalId()",
```
Comment (rhouthuijzen) [inline mapping.json:175]: Is this something for the mapping or for in the powershell script? Maybe this should be moved to powershell?

**Review by rhouthuijzen — COMMENTED** [review]
Changes look fine. one question about the department script. And one about the mapping

Can you please verifiy if the issues are tested? I linked them to this Pull Request. Not sure if they are.

Reminder for the person that merges this PR/Branch. Please don't forget the release. It was outdated.

File: departments.ps1 (hunk 1 above)
Comment (Gsichtman) [reply departments.ps1]: This has been corrected

File: mapping.json (hunk 2 above)
Comment (Gsichtman) [reply mapping.json:175]: Correct!. Discussed with Mouki, the raw should be as pure as possible and calculation in the mapping

**Review by rhouthuijzen — APPROVED** [review]

**Pull request merged**